Feature 612 talon

CarlyAmar opened the pull request.

CarlyAmar committed:
CANTalon612 commit. Has constructors and many other utility methods

CarlyAmar committed:
More work done on the new Talon

CarlyAmar committed:
Got some more work done with the outputs

CarlyAmar committed:
Implemented theoretically correctly

CarlyAmar committed:
Added testing code for motor controller

CarlyAmar committed:
Did some cleaning and documentation

WardBenjamin commented:
This looks fine to me, and the Travis CI build passed. Seems fine to merge.

ZackAlfakir commented:
I want to keep it open until I test it on a robot first

Commit:
Reformatted comment to be easier to read.

Commit:
I clipped public too (oops!).  This fixes that.

Commit:
Added comments for changes.

CarlyAmar committed:
Made it a speedcontroller instead of a CANTalon. made some fixes

ZackAlfakir commented:
I think the things needed  are:
getOutput() - make sure we are getting a valid value for output
Assign a real value to MIN_INPUT and MAX_INPUT
Confirm that this works by deploying and running with a motor

CarlyAmar committed:
Documentation and some value default fixes

CarlyAmar committed:
More fixes

Commit:
Some changes to comments and things for Zack to investigate.

Commit:
Slight change to the comment formatting.

CarlyAmar committed:
Made some changes

CarlyAmar committed:
Merge branch 'master' into feature-612Talon

Conflicts:
	src/Robot.cpp
	src/Robot.h

Commit:
Updating this branch to include master's changes to make a merge into master easier after Zack continues edits.

Merge branch 'master' into feature-612Talon

Conflicts:
	src/Robot.cpp
	src/Robot.h
